File: main/src/main/java/com/sedmelluq/lavaplayer/core/source/youtube/YoutubeTrackDetailsLoader.java
package com.sedmelluq.lavaplayer.core.source.youtube;

import com.sedmelluq.lavaplayer.core.http.HttpInterface;
import com.sedmelluq.lavaplayer.core.info.track.AudioTrackInfoTemplate;

public interface YoutubeTrackDetailsLoader {
  YoutubeTrackDetails loadDetails(HttpInterface httpInterface, String videoId, AudioTrackInfoTemplate template);
}

File: main/src/main/java/lavaplayer/source/youtube/YoutubeTrackDetailsLoader.kt
package lavaplayer.source.youtube

import lavaplayer.tools.io.HttpInterface

interface YoutubeTrackDetailsLoader {
    fun loadDetails(
        httpInterface: HttpInterface,
        videoId: String,
        requireFormats: Boolean,
        sourceManager: YoutubeItemSourceManager
    ): YoutubeTrackDetails?
}
